pkg/manager/manager.go

```diff
@@ -332,20 +332,63 @@ func (m *SupportBundleManager) compressBundle() error {
 	return nil
 }
 
-func (m *SupportBundleManager) refreshNodes() error {
-	nodes, err := m.k8s.GetNodesListByLabels(m.NodeSelector)
+func (m *SupportBundleManager) getAgentPodsCreatedBy(daemonSet *appsv1.DaemonSet) (*v1.PodList, error) {
+	startTime := time.Now()
+	ticker := time.NewTicker(types.PodCreationWaitInterval)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		pods, err := m.k8s.GetPodsListByLabels(m.PodNamespace, fmt.Sprintf("app=%s", daemonSet.Name))
+		if err != nil {
+			return nil, err
+		}
+
+		if len(pods.Items) == int(daemonSet.Status.DesiredNumberScheduled) {
+			return pods, nil
+		}
+
+		// Return error when exceeds timeout
+		if time.Since(startTime) > types.PodCreationTimeout {
+			return nil, fmt.Errorf("timed out")
+		}
+	}
+	return nil, fmt.Errorf("unexpected error: stopped waiting for creating DaemonSet Pod or timing out")
+
+}
+
+func (m *SupportBundleManager) getAgentNodesIn(podList *v1.PodList) ([]*v1.Node, error) {
+	var nodes []*v1.Node
+	for _, pod := range podList.Items {
+		node, err := m.k8s.GetNodeBy(pod.Spec.NodeName)
+		if err != nil {
+			return nil, err
+		}
+
+		nodes = append(nodes, node)
+	}
+	return nodes, nil
+}
+
+func (m *SupportBundleManager) refreshNodes(agentDaemonSet *appsv1.DaemonSet) error {
+	podList, err := m.getAgentPodsCreatedBy(agentDaemonSet)
+	if err != nil {
+		return err
+	}
+
+	nodes, err := m.getAgentNodesIn(podList)
 	if err != nil {
 		return err
 	}
 
-	if len(nodes.Items) == 0 {
+	if len(nodes) == 0 {
 		return errors.New("no nodes are found")
 	}
 
 	m.expectedNodes = make(map[string]string)
+	logrus.Debugf("expected bundles from nodes: %+v", m.expectedNodes)
```

Comment:
`logrus.Debugf("Expecting bundles from nodes: %+v", nodes)` 

Is this log verbose? If yes, probably you can log `m.expectedNodes` just before the end of this function via defer.